core-web/libs/sdk/react/src/lib/next/hooks/useIsDevMode.ts

```diff
@@ -0,0 +1,39 @@
+import { useContext, useEffect, useState } from 'react';
+
+import {
+    DotCMSPageContext,
+    DotCMSPageRendererMode
+} from '@dotcms/react/next/contexts/DotCMSPageContext';
+import { getUVEState } from '@dotcms/uve';
+import { UVE_MODE } from '@dotcms/uve/types';
+
+/**
+ * A React hook that determines if the current environment is in development mode.
+ *
+ * The hook returns `true` if either:
+ *   - The context mode (or the optional `renderMode` argument) is set to 'development', or
+ *   - The application is running inside the DotCMS editor (as determined by `isInsideEditor()`).
+ *
+ * @param {DotCMSPageRendererMode} [renderMode] - Optional override for the render mode.
+ * @returns {boolean} - `true` if in development mode or inside the editor; otherwise, `false`.
+ */
+export const useIsDevMode = (renderMode?: DotCMSPageRendererMode) => {
+    const { mode } = useContext(DotCMSPageContext);
+    const effectiveMode = renderMode ?? mode;
+
+    const [isDevMode, setIsDevMode] = useState(effectiveMode === 'development');
+
+    useEffect(() => {
+        // Inside UVE we rely on the UVE state to determine if we are in development mode
+        if (getUVEState()?.mode) {
+            const isUVEInEditor = getUVEState()?.mode === UVE_MODE.EDIT;
+            setIsDevMode(isUVEInEditor);
+
+            return;
+        }
+
+        setIsDevMode(effectiveMode === 'development');
+    }, [effectiveMode]);
```

Comment:
I think you can use a `useMemo` for this

Reply:
I'll try iy

Reply:
Done!

Reply:
The `useMemo` hook works perfectly outside the UVE context. However, when attempting to use `useMemo` to check `devMode` via `getUVEState()?.mode` inside the `UVE`, the following error occurs:

![error](https://github.com/user-attachments/assets/b8883c99-ee5f-4d34-9ebf-3bb3b3b37887)

This happens because `useMemo` is executed before the DOM is ready. Then, when the DOM becomes available, it re-triggers, causing a hydration error.